client-lite/src/download/download.cpp

```diff
@@ -432,20 +444,135 @@ UINT Download::_MaxNoProgressIntervals() const
 {
     if (_noProgressTimeout == _unsetTimeout)
     {
-        if (!_mccHost.empty() && _mccManager.NoFallback() && HttpAgent::IsClientError(_httpStatusCode))
+        return g_progressTrackerMaxNoProgressIntervals;
+    }
+
+    auto maxNoProgressIntervals = std::round(std::chrono::duration<double>(_noProgressTimeout) / g_progressTrackerCheckInterval);
+    DO_ASSERT(maxNoProgressIntervals >= 1);
+    return static_cast<UINT>(maxNoProgressIntervals);
+}
+
+std::string Download::_UpdateConnectionTypeAndGetUrl(bool retryAfterFailure)
+{
+    auto newConnectionType = _connectionType;
+    if (retryAfterFailure)
+    {
+        DO_ASSERT(_connectionType != ConnectionType::None);
+
+        _fAllowMcc = !_FallbackFromMccDue();
+
+        if (_fAllowMcc && !_mccFallbackDue)
```

Comment:
Can this be rewritten as:

if (!_FallbackfromMccDue())

Since _mccFallbackDue is already checked in 462

Reply:
Reworked this a bit.